### PR TITLE
task 3 Anton Khlebko

### PR DESCRIPTION
**Pull request by AKhlebko**
HomeWork 3

### CODE REVIEW

File: DEV-3/DEV-3/RadixConverter.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Numerics;
+using System.Text;
+
+namespace taskDEV3
+{
+    /// <summary>
+    /// Class which stores decimal number's form
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/RadixConverter.cs]: what do you mean switch? Convert

File: DEV-3/DEV-3/RadixConverter.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    /// and can switch it into some other radixs from 2 to 20
+    /// </summary>
+    class RadixConverter
+    {
+        BigInteger decimalNumber;
+        short newRadix;
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/RadixConverter.cs]: Why do you make numberInNewRadix as a class field? no need

File: DEV-3/DEV-3/RadixConverter.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        StringBuilder numberInNewRadix;
+        
+        public RadixConverter(BigInteger pDecimalNumber, short pNewRadix)
+        {
+            decimalNumber = pDecimalNumber;
+            newRadix = pNewRadix;
+            numberInNewRadix = new StringBuilder();
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/RadixConverter.cs]: has to be local value

File: DEV-3/DEV-3/RadixConverter.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            BigInteger bufferDecimalNumber = decimalNumber;
+            while (bufferDecimalNumber > 0)
+            {
+                short modulo = (short)(bufferDecimalNumber % newRadix);
+                if (modulo < 10)
+                {
+                    numberInNewRadix.Insert(0, modulo);
+                }
+                else
+                {
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/RadixConverter.cs]: What is 55? Explain and use const value for it

File: DEV-3/DEV-3/InputDataHandler.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+        /// <summary>
+        /// either unpack args into decimalNumer and newRadix
+        /// if there are no format mistakes
+        /// or places firstFindMistakeType mistake's type
+        /// </summary>
+        /// <returns>
+        /// True if data format is right
+        /// False if it's not
+        /// </returns>
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/InputDataHandler.cs]: Unclear name and purpose of this method

File: DEV-3/DEV-3/InputDataHandler.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        public short newRadix { get; set; }
+        string[] data;
+        InputMistakeType firstFoundMistakeType;
+
+        public InputDataHandler(string[] args)
+        {
+            data = args;
+        }
+
+        /// <summary>
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/InputDataHandler.cs]: What do you mean "unpack"?

File: DEV-3/DEV-3/InputDataHandler.cs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        /// <returns>
+        /// True if data format is right
+        /// False if it's not
+        /// </returns>
+        public bool UnPackAndCheckDataFormat()
+        {
+            bool response = true;
+            BigInteger bufferDecimal;
+            short bufferRadix;
+            if (data.Length != 2)
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/InputDataHandler.cs]: Use exception

File: DEV-3/DEV-3/InputDataHandler.cs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        WrongDecimalFormat,
+        WrongDecimalSign,
+        WrongRadixFormat,
+        WrongRadixValue
+    }
+
+    /// <summary>
+    /// Class for parsing console args and
+    /// checking either their format is acceptable or not
+    /// </summary>
```
Comment (IgorKheidorov) [inline DEV-3/DEV-3/InputDataHandler.cs]: Class for console args checking is not Handler

File: DEV-3/DEV-3/InputDataHandler.cs (hunk 6 above)
Comment (AKhlebko) [reply DEV-3/DEV-3/InputDataHandler.cs]: I mean unpack it from console "string[] args" into decimalNumber and newRadix. In my opinion unpack is rather proper name here

File: DEV-3/DEV-3/InputDataHandler.cs (hunk 7 above)
Comment (AKhlebko) [reply DEV-3/DEV-3/InputDataHandler.cs]: I did, just adding some fixes before commiting

File: DEV-3/DEV-3/RadixConverter.cs (hunk 3 above)
Comment (AKhlebko) [reply DEV-3/DEV-3/RadixConverter.cs]: If I make it local I'll break single responsibility rule. Or i'll have to rename method into "ConverNumberRadixAndPrintIt" quite long name for the method

File: DEV-3/DEV-3/EntryPoint.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
+﻿namespace taskDEV3
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            InputDataChecker preparer = new InputDataChecker(args);
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/EntryPoint.cs]: Missed exception handling

File: DEV-3/DEV-3/InputDataChecker.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Numerics;
+
+namespace taskDEV3
+{
+    /// <summary>
+    /// Class checks input data format and 
+    /// if it's right prepare data for transfering into RadixConverter instance
+    /// </summary>
+    class InputDataChecker
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/InputDataChecker.cs]: checker cannot prepare - single responsibility violation

File: DEV-3/DEV-3/InputDataChecker.cs [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        string[] data;
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/InputDataChecker.cs]: invalid field name

File: DEV-3/DEV-3/InputDataChecker.cs [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+            { 
+                decimalNumber = BigInteger.Parse(data[0]);
+                newRadix = int.Parse(data[1]); 
+                if (newRadix < 2 || newRadix > 20)
+                {
+                    response = false;
+                    throw new Exception("Radix value is out of correct range.");
+                }
+            }
+            catch (IndexOutOfRangeException)
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/InputDataChecker.cs]: invalid level of exception handling

File: DEV-3/DEV-3/ConsoleDataFormatChecker.cs [hunk 13]
```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Numerics;
+
+namespace taskDEV3
+{
+    /// <summary>
+    /// Class has method for checking
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/ConsoleDataFormatChecker.cs]: Wrong class description, it must describe what and how class implements, not the methods it has

File: DEV-3/DEV-3/ConsoleDataFormatChecker.cs [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        /// <summary>
+        /// Checks commandLineArgs' format
+        /// </summary>
+        /// <returns>
+        /// Returns true if data format is right
+        /// Returns false if it's not
+        /// </returns>
+        public bool CheckInputDataFormat()
```
Comment (AlexanderSoroka) [inline DEV-3/DEV-3/ConsoleDataFormatChecker.cs]: completely invalid exception handling logic - awful mix of return code and exception approach

**Pull request merged**